apiserver/src/main/java/org/dependencytrack/tasks/BomUploadProcessingTask.java

```diff
@@ -880,10 +881,12 @@ private static long deleteComponentsById(final QueryManager qm, final Collection
         pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.Analysis WHERE :ids.contains(component.id)").execute(componentIds);
         pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.ViolationAnalysisComment WHERE :ids.contains(violationAnalysis.component.id)").execute(componentIds);
         pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.ViolationAnalysis WHERE :ids.contains(component.id)").execute(componentIds);
-        pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.DependencyMetrics WHERE :ids.contains(component.id)").execute(componentIds);
         pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.FindingAttribution WHERE :ids.contains(component.id)").execute(componentIds);
         pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.PolicyViolation WHERE :ids.contains(component.id)").execute(componentIds);
         pm.newQuery(Query.JDOQL, "DELETE FROM org.dependencytrack.model.IntegrityAnalysis WHERE :ids.contains(component.id)").execute(componentIds);
+
+        withJdbiHandle(handle -> handle.attach(MetricsDao.class).deleteDependencyMetricsByComponentIds(componentIds.stream().toList()));
+
```

Comment:
I think you can actually remove everything except `pm.newQuery(Component.class, ":ids.contains(id)").deletePersistentAll(componentIds);`. The foreign keys with `ON DELETE CASCADE` should take care of deleting everything else.

Seems we missed this back when we added the foreign key constraints.